### DISCUSSION TITLE
Is there an efficient way to do tensor-scalar addition?

### DISCUSSION THREAD

**Discussion by balisujohn** [ggml-org/ggml#811] [Q&A] [answered]
Right now the most efficient way I know how to do it is create tensor with the same size as the target tensor, then manually set its all its values to the scalar value before running the computational graph, then running the computational graph and adding the the tensor holding the scalar to the target tensor. This seems brutally inefficient computationally; is there a better way to add a scalar to each value in a tensor?

**Comment by slaren** [comment, marked as answer]
The add operation support broadcasting, so you only really need a 1 element tensor.
